### ISSUE TITLE
Aborted with core dump

### ISSUE THREAD

**Issue by vishalvvr** [nuspell/nuspell#111]
- Bug reports
# Reporting bugs

| key                                       | value |
|-------------------------------------------|-------|
| Operating system, distribution, version = |    fedora-34   |
| Nuspell version                         = |   4.2.0   |
| Dictionary, package name, version       = |       |
| Command-line tool or GUI application    = |       |

#### Have a text file with incorrect string "hella my nama is fedora"
$ nuspell -d en_US text.txt     
terminate called after throwing an instance of 'std::system_error'
  what():  Unknown error -1
Aborted (core dumped)

## Bugged behavior (output)
Aborted (core dumped)

## Expected behavior (output)
Display correct word suggestion for the strings "hella" & "nama".

## Environment
$ echo $XDG_DATA_DIRS/myspell
/home/vishal/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share/:/usr/share//myspell

$ echo $XDG_DATA_DIRS/hunspell
/home/vishal/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share/:/usr/share//hunspell

$ nuspell -D
Search paths:
/home/vishal/.local/share/hunspell
/home/vishal/.local/share/flatpak/exports/share/hunspell
/var/lib/flatpak/exports/share/hunspell
/usr/local/share//hunspell
/usr/share//hunspell
/home/vishal/.local/share/flatpak/exports/share/myspell
/var/lib/flatpak/exports/share/myspell
/usr/local/share//myspell
/usr/share//myspell
.
Available dictionaries:
en_AG           /usr/share//myspell/en_AG
en_AU           /usr/share//myspell/en_AU
en_BS           /usr/share//myspell/en_BS
en_BW           /usr/share//myspell/en_BW
en_BZ           /usr/share//myspell/en_BZ
en_CA           /usr/share//myspell/en_CA
en_DK           /usr/share//myspell/en_DK
en_GB           /usr/share//myspell/en_GB
en_GH           /usr/share//myspell/en_GH
en_HK           /usr/share//myspell/en_HK
en_IE           /usr/share//myspell/en_IE
en_IN           /usr/share//myspell/en_IN
en_JM           /usr/share//myspell/en_JM
en_MW           /usr/share//myspell/en_MW
en_NA           /usr/share//myspell/en_NA
en_NG           /usr/share//myspell/en_NG
en_NZ           /usr/share//myspell/en_NZ
en_PH           /usr/share//myspell/en_PH
en_SG           /usr/share//myspell/en_SG
en_TT           /usr/share//myspell/en_TT
en_US           /usr/share//myspell/en_US
en_ZA           /usr/share//myspell/en_ZA
en_ZM           /usr/share//myspell/en_ZM
en_ZW           /usr/share//myspell/en_ZW

### Note: 
- This issue was also reproducible in nuspell 5.0.0 [build](https://copr.fedorainfracloud.org/coprs/vishalvvr/nuspell/build/2252356)
- Appimage works fine.
- Is this dir path correct  "/usr/share//myspell" & " /usr/share//hunspell" (output : nuspell -D)?

**Comment by dimztimz** [comment]
This bug is specific only to Fedora. I was not able to reproduce it on Ubuntu but I can reproduce it on Fedora 34 (inside docker).

The bug is not in nuspell but in its dependencies, ICU (libicu). The following minimal code triggers the bug:

```cpp
#include <unicode/locid.h>

int main()
{
        icu::Locale l; // just calling the default constructor fails
}
```
To trigger the bug I need to compile like this:

```bash
g++ test.cpp -g `pkg-config icu-uc --libs` 
```
The bug is somewhere in code about using threads which suggests the threading library Pthread is not used. If I compile like this

```bash
g++ test.cpp -g `pkg-config icu-uc --libs` -lpthread
```
Then the bug is gone. You should open a bug on the side of Fedora about the package `libicu` and fix it so it builds and links with pthread.

**Comment by dimztimz** [comment]
Just tested under Fedora rawhide and the bug is not present. It seems to me this was caused by a bug in GCC 11.0 (considered a pre-release) which got fixed in 11.1 (stable release). 

For more info see: https://bugzilla.redhat.com/show_bug.cgi?id=1954178 https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99341

Fedora 34 used GCC 11.0.x at some point and probably that causes the bug. Fedora rawhide is already on GCC 11.1 and seems like there isn't any bug there. I'm closing this.
